src/shadow-embed.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ampdocFor} from './ampdoc';
+import {dev} from './log';
+import {extensionsFor} from './extensions';
+import {copyRuntimStylesToShadowRoot} from './styles';
+
+
+/**
+ * @param {!Element} hostElement
+ * @param {!Array<string>} extensionIds
+ * @return {!ShadowRoot}
+ */
+export function createShadowEmbedRoot(hostElement, extensionIds) {
+  dev.assert(!hostElement.shadowRoot,
+      'Shadow root has already been created on %s', hostElement);
+  const shadowRoot = hostElement.createShadowRoot();
+  shadowRoot.AMP = {};
+
+  const win = hostElement.ownerDocument.defaultView;
+  const extensions = extensionsFor(win);
+  const ampdocService = ampdocFor(win);
+  const ampdoc = ampdocService.getAmpDoc(hostElement);
+
+  // Instal runtime CSS.
+  copyRuntimStylesToShadowRoot(ampdoc, shadowRoot);
+
+  // Install extensions.
```

Comment:
Could you point me to where that selection is made?

On Thu, Jul 21, 2016 at 9:57 AM, Dima Voytenko <notifications@github.com>
wrote:

> In src/shadow-embed.js
> <https://github.com/ampproject/amphtml/pull/4148#discussion_r71744089>:
>
> > + */
> > +export function createShadowEmbedRoot(hostElement, extensionIds) {
> > +  dev.assert(!hostElement.shadowRoot,
> > +      'Shadow root has already been created on %s', hostElement);
> > +  const shadowRoot = hostElement.createShadowRoot();
> > +  shadowRoot.AMP = {};
> > +
> > +  const win = hostElement.ownerDocument.defaultView;
> > +  const extensions = extensionsFor(win);
> > +  const ampdocService = ampdocFor(win);
> > +  const ampdoc = ampdocService.getAmpDoc(hostElement);
> > +
> > +  // Instal runtime CSS.
> > +  copyRuntimStylesToShadowRoot(ampdoc, shadowRoot);
> > +
> > +  // Install extensions.
>
> Currently it install only those used in the embed. Why would we install
> all?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ampproject/amphtml/pull/4148/files/b10e1bb03a5b8c00b481651a18c78d71ba7485b9#r71744089>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAFeTxGlaHHefagoppzTUDURMDtcsRjtks5qX6T8gaJpZM4JRV82>
> .
>